src/rgw/rgw_bucket.cc

```diff
@@ -1470,7 +1470,8 @@ int RGWBucketAdminOp::limit_check(RGWRados *store,
 	}
 
 	num_shards = info.num_shards;
-	uint64_t objs_per_shard = num_objects / num_shards;
+	uint64_t objs_per_shard = (num_shards) ?
+	  num_objects/num_shards : std::max(uint64_t(0), num_objects);
```

Comment:
ok, yeah, it's just num_objects;  as noted, on downstream, I see info.num_shareds == 0 when there is no sharding